File: GomokuAIJavaProject/AIGomoku/Controller.java
package AIGomoku;
import java.util.Scanner;

public class Controller {
	
	
	private static Controller instance;
	private Scanner scan;
	
	public static Controller getInstance() {
		if(instance == null) {
			instance = new Controller();
		}
		return instance;
	}
	
	
	private Controller() {
		scan = new Scanner(System.in);
	}

	public int getPos() {
		
		int nb = scan.nextInt()-1;
		while(nb < 0 || nb > Board.BOARD_SIZE-1) {
			System.out.println("Select a number");
			nb = scan.nextInt()-1;
		}
		
		return nb;
	}
}

File: CPP/src/Controller.cpp
#include "Controller.hpp"

unsigned int Controller::getPos()
{
    int nb = -1;
    std::cin >> nb;
    nb--;
    while(nb < 0 || nb > (int)Board::BOARD_SIZE-1) {
        std::cout << "Select a number" << std::endl;
        std::cin.clear();
        std::cin >> nb;
        nb--;
    }
    
    return nb;
}
